environment.yml

```diff
@@ -0,0 +1,24 @@
+name: detectron2-env
+channels:
+  - pytorch
+  - defaults
+dependencies:
+  - python==3.7
+  - pip
+  - cython
+  - pytorch::pytorch
+  - pytorch::torchvision
+  - pyyaml==5.1
+  - gcc_linux-64 ## gcc, g++ only needed for source installs of Detectron
+  - gxx_linux-64
+  - git
+  - wget
+  - unzip
+  - matplotlib
+  - conda-forge::pycocotools
+  - pip:
+    - opencv-contrib-python
+    ## install Detectron v2 itself:
+    #- git+https://github.com/facebookresearch/detectron2.git ## source build (needed if you use custom compiled pytorch etc) 
+    #or run: pip install detectron2 -f https://dl.fbaipublicfiles.com/detectron2/wheels/cu101/index.html # replace cu101 with cu100,90, or cpu. 
```

Comment:
This yaml config does not specify a cuda version for pytorch/torchvision, etc, so my guess is that it will not always pick the cuda version that the user had on the machine, which is going to cause problems.

Reply:
> This yaml config does not specify a cuda version for pytorch/torchvision, etc, so my guess is that it will not always pick the cuda version that the user had on the machine

actually that's the point, afaik conda will try to resolve and pick the version it found on the machine. I could easily force the `10.1` version in those deps, but I chose to let it open so the user can have whichever they use.

Reply:
> afaik conda will try to resolve and pick the version it found on the machine

I am not a conda user but this surprises me. Actually I doubt it's actually true because guessing what cuda version to use (especially if there are multiple cuda installed in the system) is not a reasonable thing for conda to do. Unless we know how conda selects the cuda version to use, we should allow users to specify which version to use.

Forcing a 10.1 version is only going to cause more environment issues for users without 10.1.

Reply:
I had just cuda 10.1 installed and this seemed to work, I'm testing with multiple cuda versions (although that might not be a typical config)

Reply:
I think pytorch defaults to use 10.1 so it's reasonable this works on 10.1.
I'm more interested to know whether it would automatically install the cuda10 version of pytorch (which should be confirmed by `print(torch.__config__.show())`)

Reply:
> to know whether it would automatically install the cuda10 version of pytorch (which should be confirmed by print(torch.__config__.show())) 

I was going to test with cuda-10-2, so should I rather try cuda-10-0?

Reply:
Pytorch doesn't have packages for cuda 10.2: https://anaconda.org/pytorch/pytorch/files

So if you try cuda 10.2 and it still successfully installs - then it must be doing things incorrectly.

Reply:
> So if you try cuda 10.2 and it still successfully installs - then it must be doing things incorrectly 

yep, you're right, conda does not check which/if CUDA is installed at all on the "host" system. So the env creates. 

OT: funnily, this still runs with cuda-10-2, even though pytorch and the wheel package are for 10.1 only. 

How can we go about this, then?